### DISCUSSION TITLE
Advanced breaking change notice for applications and groups Bicep types

### DISCUSSION THREAD

**Discussion by dkershaw10** [microsoftgraph/msgraph-bicep-types#65] [Announcements]
Folks,

Initially we wanted to be a bit more consistent with ARM types, where _name_ represents the unique name identifier, but we think this mapping is causing more confusion. when it's mapped to something else in the API (as seen in [groups resource with existing keyword · Issue #39](https://github.com/microsoftgraph/msgraph-bicep-types/issues/39) and [Existing on Office 365 groups · Issue #61](https://github.com/microsoftgraph/msgraph-bicep-types/issues/61)).  So we're renaming the `applications` and `groups` Bicep types' _name_ property to _uniqueName_ to be consistent with the Graph API schema. 
 
We'll roll out a new Bicep version, which will have these changes. Once released, please update to use the new Bicep version in VS Code, and make the updates in your Bicep template files.  The Graph Bicep extension service will continue to support existing Bicep template files that contain the _name_ property (for `applications` and `groups`) for **two weeks** after the new Bicep version is made available.
 
>**NOTE**: After the two weeks, the service will only support _uniqueName_, and any existing files that have not been updated will fail deployments.

**Comment by slavizh** [comment]
ok. sounds reasonable to me. Would this apply to other resources as well in the future part of the graph or currently will be only for applications and groups and some others that might fall in the same category?

**Comment by dkershaw10** [comment]
I think we will adopt this pattern (moving forward) for any new Graph types that we add. We have internal guidelines for teams to add a client provided key called `uniqueName` if the Graph resource doesn't already have a client provided key.
